### PR TITLE
add new testing setup

### PR DESCRIPTION
**Pull request by JeanMertz**
* be more explicit in tests
* dynamically define resources
* test all the possible CRUD situations

### CODE REVIEW

File: lib/eipiai/webmachine/resources/singular.rb [hunk 1]
```diff
@@ -42,7 +42,7 @@ def from_json
     private
 
     def params
-      @merged_params ||= super.merge('uid' => object_uid)
+      @merged_params ||= super
```
Comment (jurriaan) [inline lib/eipiai/webmachine/resources/singular.rb]: Nothing merged here, right?

File: lib/eipiai/models/representable.rb [hunk 2]
```diff
@@ -33,7 +33,7 @@ def represented
     # @param [Hash] options to be used inside the representer
     # @return [Hash] hash representation of the object
     #
-    def to_h(**options)
+    def to_h(options = {})
```
Comment (jurriaan) [inline lib/eipiai/models/representable.rb]: Why?

File: lib/eipiai/models/representable.rb [hunk 3]
```diff
@@ -48,7 +48,7 @@ def to_h(**options)
     # @param [Hash] hash to be used to set the object attributes
     # @return [Object] the object itself
     #
-    def from_hash(**hash)
+    def from_hash(hash = {})
```
Comment (jurriaan) [inline lib/eipiai/models/representable.rb]: Why?

File: lib/eipiai/models/representable.rb (hunk 2 above)
Comment (JeanMertz) [inline lib/eipiai/models/representable.rb]: See commit message. Because of string-based keys.

File: lib/eipiai/models/representable.rb (hunk 3 above)
Comment (JeanMertz) [inline lib/eipiai/models/representable.rb]: see above

**Comment by jurriaan** [comment]
:+1:

**Pull request merged**